packages/@aws-cdk/core/lib/names.ts

```diff
@@ -36,5 +38,26 @@ export class Names {
     return components.length > 0 ? makeUniqueId(components) : '';
   }
 
+  /**
+   * Returns a CloudFormation-compatible unique identifier for a construct based
+   * on its path. If the node is a Stack, the stack name is used for that path component instead.
+   * The identifier includes a human readable portion rendered from the path components and a hash suffix.
+   * @param construct The construct
+   * @param maxUniqueIdLength The maximum length of the unique id
+   * @returns a unique id based on the construct path
+   */
+  public static uniqueIdWithStackName(construct: Construct, maxUniqueIdLength: number = 256): string {
+    const node = Node.of(construct);
```

Comment:
Maybe also add a comment to `uniqueId` to say something like "`uniqueId` is not unique if multiple copies of the stack are deployed. Prefer using `uniqueIdWithStackName()`."

Reply:
Added this to the comments.